assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -145,74 +132,69 @@ export class SpreadsheetFrontend extends Component {
-            placeholder={__('Search data', 'planet4-blocks')}
-          />
-          <div className="table-wrapper">
-            <table className={`spreadsheet-table is-color-${this.props.color ?? 'grey'}`}>
-              <thead>
-                <tr>
-                  {
-                    headers.map((cell, index) => (
-                      <th
-                        className={(
-                          index === this.state.sortColumnIndex ?
-                            `spreadsheet-sorted-by sort-${this.state.sortDirection}` :
-                            ''
-                        )}
-                        onClick={() => {
-                          this.onHeaderClick(index);
-                        }}
-                        key={index}
-                        scope="col"
-                        title={cell}>
-                        <button>
-                          { cell }
-                          <ArrowIcon />
-                        </button>
-                      </th>
-                    ))
-                  }
-                </tr>
-              </thead>
-              <tbody>
+  };
+
+  const headers = spreadsheetData ? spreadsheetData.header : placeholderData.header;
+
+  return (
+    <>
+      <section className={`block block-spreadsheet ${className ?? ''}`}>
+        <input className="spreadsheet-search form-control"
+          type="text"
+          value={searchText}
+          onChange={event => setSearchText(event.target.value)}
+          placeholder={__('Search data', 'planet4-blocks')}
+        />
+        <div className="table-wrapper">
+          <table className={`spreadsheet-table is-color-${color ?? 'grey'}`}>
+            <thead>
+              <tr>
                 {
-                  this.state.loading === true ?
-                    <tr>
-                      <td colSpan="99999">
-                        <div className="spreadsheet-loading">{ __('Loading spreadsheet data…', 'planet4-blocks') }</div>
-                      </td>
-                    </tr> :
-                    this.renderRows()
+                  headers.map((cell, index) => (
+                    <th
+                      className={(
+                        index === sortColumnIndex ?
+                          `spreadsheet-sorted-by sort-${sortDirection}` :
+                          ''
+                      )}
+                      onClick={() => onHeaderClick(index)}
+                      key={index}
+                      scope="col"
+                      title={cell}>
+                      <button>
+                        {cell}
+                        <ArrowIcon />
+                      </button>
+                    </th>
+                  ))
                 }
-              </tbody>
-            </table>
-          </div>
-        </section>
-      </Fragment>
-    );
-  }
-}
+              </tr>
+            </thead>
+            <tbody>
+              {
+                loading === true ?
```

Comment:
I don't see necessary to evaluate if the value is `=== true`.

Reply:
I just kept the previous implementation, but I can change it